File: typescipt/src/compiler/cSharp/builtInFunctions/noArgumentFunctionCall.ts


internal abstract class NoArgumentFunctionCall : FunctionCall {
   public NoArgumentFunction Function

   protected abstract string ClassName
   protected abstract string MethodName

   protected NoArgumentFunctionCall(NoArgumentFunction function) : base(function) {
     Function = function;
   }

   public override customMethodSyntax(context: ICompileFunctionContext): MemberDeclarationSyntax {
     return null;
   }

   public override callExpressionSyntax(context: ICompileFunctionContext): ExpressionSyntax {
     return SyntaxFactory.InvocationExpression(
       SyntaxFactory.MemberAccessExpression(
         SyntaxKind.SimpleMemberAccessExpression,
         SyntaxFactory.IdentifierName(ClassName),
         SyntaxFactory.IdentifierName(MethodName)));
   }
}

File: dotnet/Lexy/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/NoArgumentFunctionCall.cs
using Lexy.Compiler.Language.Expressions.Functions;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal abstract class NoArgumentFunctionCall : FunctionCall<NoArgumentFunction>
{
    protected abstract string ClassName { get; }
    protected abstract string MethodName { get; }

    public override ExpressionSyntax CallExpressionSyntax(NoArgumentFunction _)
    {
        return SyntaxFactory.InvocationExpression(
            SyntaxFactory.MemberAccessExpression(
                SyntaxKind.SimpleMemberAccessExpression,
                SyntaxFactory.IdentifierName(ClassName),
                SyntaxFactory.IdentifierName(MethodName)));
    }
}